### PR TITLE
Add travis configuration and fix flake8 failures

### PR DESCRIPTION
**Pull request by prajakta-gokhale**
* Add `travis.yml` to setup Travis CI
* Make `test_flake8` similar to `launch_ros_sandbox`
* Fix test_flake8 failures

Fixes https://github.com/ros-security/aws-roadmap/issues/34.

Signed-off-by: Prajakta Gokhale <prajaktg@amazon.com>

### CODE REVIEW

**Comment by emersonknapp** [comment]
This looks pretty good, but should the travis be passing?

**Comment by prajakta-gokhale** [comment]
> This looks pretty good, but should the travis be passing?

I was expecting it to pass, since the tests passed locally using the same commands. I'll double check tomorrow to see why it failed.

**Comment by thomas-moulard** [comment]
```
==================================== ERRORS ====================================
_____________ ERROR at setup of test_sysroot_compiler_constructor ______________
file /home/travis/ros2_latest_cross_compile_ws/src/cross_compile/test/test_sysroot_compiler.py, line 97
  def test_sysroot_compiler_constructor(
E       fixture 'tmp_path' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, docker_config, doctest_namespace, mock, mocker, monkeypatch, platform_config, pytestconfig, record_xml_property, recwarn, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.
/home/travis/ros2_latest_cross_compile_ws/src/cross_compile/test/test_sysroot_compiler.py:97
_______________ ERROR at setup of test_write_cc_build_setup_file _______________
file /home/travis/ros2_latest_cross_compile_ws/src/cross_compile/test/test_sysroot_compiler.py, line 110
  def test_write_cc_build_setup_file(platform_config, docker_config, tmp_path):
E       fixture 'tmp_path' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, docker_config, doctest_namespace, mock, mocker, monkeypatch, platform_config, pytestconfig, record_xml_property, recwarn, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.
/home/travis/ros2_latest_cross_compile_ws/src/cross_compile/test/test_sysroot_compiler.py:110
______________ ERROR at setup of test_write_cc_system_setup_file _______________
file /home/travis/ros2_latest_cross_compile_ws/src/cross_compile/test/test_sysroot_compiler.py, line 121
  def test_write_cc_system_setup_file(platform_config, docker_config, tmp_path):
E       fixture 'tmp_path' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, docker_config, doctest_namespace, mock, mocker, monkeypatch, platform_config, pytestconfig, record_xml_property, recwarn, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.
/home/travis/ros2_latest_cross_compile_ws/src/cross_compile/test/test_sysroot_compiler.py:121
=================================== FAILURES ===================================
_________________________________ test_flake8 __________________________________
test/test_flake8.py:61: in test_flake8
    assert not report.total_errors, \
E   AssertionError: flake8 reported 5 errors
E   assert not 5
E    +  where 5 = <flake8.api.legacy.Report object at 0x7fc858ee1c88>.total_errors
----------------------------- Captured stdout call -----------------------------
5     E501 line too long (92 > 79 characters)
----------------------------- Captured stderr call -----------------------------
ERROR:flake8.api.legacy:Could not update option "extend_ignore"
legacy.py                   46 ERROR    Could not update option "extend_ignore"
ERROR:flake8.api.legacy:Could not update option "extend_ignore"
legacy.py                   46 ERROR    Could not update option "extend_ignore"
/home/travis/ros2_latest_cross_compile_ws/src/cross_compile/cross_compile/sysroot_compiler.py:246:80: E501 line too long (86 > 79 characters)
        logger.info('Fetching sysroot base image: %s', self._docker_config.base_image)
                                                                               ^
/home/travis/ros2_latest_cross_compile_ws/src/cross_compile/cross_compile/sysroot_compiler.py:279:80: E501 line too long (88 > 79 characters)
                    'Error building sysroot image. The following error was caught:\n%s',
                                                                               ^
/home/travis/ros2_latest_cross_compile_ws/src/cross_compile/cross_compile/sysroot_compiler.py:281:80: E501 line too long (87 > 79 characters)
                raise docker.errors.BuildError(reason=error_line, build_log=error_line)
                                                                               ^
/home/travis/ros2_latest_cross_compile_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:92:80: E501 line too long (92 > 79 characters)
        help='Base Docker image to use for building the sysroot. Ex. arm64v8/ubuntu:bionic')
                                                                               ^
/home/travis/ros2_latest_cross_compile_ws/src/cross_compile/cross_compile/ros2_cross_compile.py:98:80: E501 line too long (82 > 79 characters)
        help="Docker's network_mode parameter to use for all Docker interactions")
                                                                               ^
flake8 reported 5 errors
------------------------------ Captured log call -------------------------------
legacy.py                   46 ERROR    Could not update option "extend_ignore"
legacy.py                   46 ERROR    Could not update option "extend_ignore"
================= 1 failed, 2 passed, 3 error in 0.64 seconds ==================
---
Finished <<< cross_compile [1.85s]	[ with test failures ]
]0;colcon test [1/1 done] [0 ongoing]
Summary: 1 package finished [3.85s]
  1 package had test failures: cross_compile
The command "colcon test --event-handlers console_cohesion+ --packages-select cross_compile --return-code-on-test-failure" exited with 1.
```

Usually those discrepancies happen if you're running flake8 from the command line instead of using `colcon test`

**Comment by prajakta-gokhale** [comment]
> The fixture problem is more concerning to me... maybe a missing import?

Looks like the version of pytest used to run the tests doesn't support `tmp_path` fixture. Changed it to `tmpdir` which is supported.

**Comment by thomas-moulard** [comment]
Seems like your attempt to break the build was unsuccessful? (Break travis build is green)

**Comment by prajakta-gokhale** [comment]
> Seems like your attempt to break the build was unsuccessful? (Break travis build is green)

I realized right after pushing that commit that it won't work because CI is still only running on `master`. Testing CI by breaking the build will have to be done outside of this PR.

File: package.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>cross_compile</name>
-  <version>0.0.1</version>
+  <version>0.0.2</version>
```
Comment (thomas-moulard) [inline package.xml]: unrelated?

**Comment by emersonknapp** [comment]
@thomas-moulard not ready for review, I will ping you when it is

File: .travis/run_travis_build [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
```
Comment (thomas-moulard) [inline .travis/run_travis_build]: `set -euo pipefail`

File: .travis/run_travis_build [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+mkdir -p ~/ros2_latest_cross_compile_ws/src
+cd ~/ros2_latest_cross_compile_ws || exit 1
```
Comment (thomas-moulard) [inline .travis/run_travis_build]: drop exit 1

File: .travis/run_travis_build [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  cross_compile:
+    type: git
+    url: "https://github.com/${TRAVIS_REPO_SLUG}.git"
+    version: "${BRANCH}"
+EOF
+
+# RTI_NC_LICENSE_ACCEPTED avoids the interactive prompt asking to accept the RTI Connext license.
+# For "latest" builds, rosdep often misses some keys, adding "|| true", to
+# ignore those failures, as it is often non-critical.
+RTI_NC_LICENSE_ACCEPTED=yes rosdep install -r --from-paths src --ignore-src --rosdistro eloquent -y || true
```
Comment (thomas-moulard) [inline .travis/run_travis_build]: Need to install RTI Connext outside of this IMHO

File: .travis/run_travis_build [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+python3 -m pip install \
+    wheel \
```
Comment (thomas-moulard) [inline .travis/run_travis_build]: alphasort
